### PR TITLE
Add gcp-ipi-proxy-etcd-encryption-workflow

### PR DESCRIPTION
**Pull request by wangke19**
Add gcp-ipi-proxy-etcd-encryption-workflow for Jira issue [OCPQE-12209](https://issues.redhat.com//browse/OCPQE-12209)

### CODE REVIEW

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/proxy/etcd-encryption/deprovision/cucushift-installer-rehearse-gcp-ipi-proxy-etcd-encryption-deprovision-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+chain:
+  as: cucushift-installer-rehearse-aws-ipi-proxy-workers-rhel8-etcd-encryption-deprovision
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/proxy/etcd-encryption/deprovision/cucushift-installer-rehearse-gcp-ipi-proxy-etcd-encryption-deprovision-chain.yaml:2]: Sounds like the chain name is wrong.

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/proxy/etcd-encryption/deprovision/cucushift-installer-rehearse-gcp-ipi-proxy-etcd-encryption-deprovision-chain.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  steps:
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/proxy/etcd-encryption/deprovision/cucushift-installer-rehearse-gcp-ipi-proxy-etcd-encryption-deprovision-chain.yaml:3]: And the following steps are wrong, which are specific for aws

Comment (wangke19) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/proxy/etcd-encryption/deprovision/cucushift-installer-rehearse-gcp-ipi-proxy-etcd-encryption-deprovision-chain.yaml:3]: Yeah, fixed.

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/proxy/etcd-encryption/deprovision/cucushift-installer-rehearse-gcp-ipi-proxy-etcd-encryption-deprovision-chain.yaml (hunk 1 above)
Comment (wangke19) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/proxy/etcd-encryption/deprovision/cucushift-installer-rehearse-gcp-ipi-proxy-etcd-encryption-deprovision-chain.yaml]: Fixed.

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/proxy/etcd-encryption/deprovision/cucushift-installer-rehearse-gcp-ipi-proxy-etcd-encryption-deprovision-chain.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+chain:
+  as: cucushift-installer-rehearse-gcp-ipi-proxy-etcd-encryption-deprovision
+  steps:
+  - chain: gather-network
+  - ref: gather-gcp-console
+  - chain: gather
+  - ref: gather-proxy
+  - ref: ipi-deprovision-deprovision
+  - ref: gcp-deprovision-stacks
+  - ref: gcp-deprovision-s3buckets
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/proxy/etcd-encryption/deprovision/cucushift-installer-rehearse-gcp-ipi-proxy-etcd-encryption-deprovision-chain.yaml:10]: I did not see these ref are existing.

Comment (wangke19) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/proxy/etcd-encryption/deprovision/cucushift-installer-rehearse-gcp-ipi-proxy-etcd-encryption-deprovision-chain.yaml]: Yes, I got the wrong copy from aws chain, just fixed.

**Comment by wangke19** [comment]
Seems something wrong with GCP installation with latest 4.12, checked the install log,
```
ClusterVersion: Installing "4.12.0-0.nightly-2022-10-05-053337" for 51 minutes: Unable to apply 4.12.0-0.nightly-2022-10-05-053337: an unknown error has occurred: MultipleErrors
```

**Comment by jianlinliu** [comment]
I guess it is because a 4.12 know bug - https://issues.redhat.com/browse/OCPBUGS-2122, for the workflow, it looks good to me, I guess you can change the test job onto 4.11, once it get passed, we can merge this PR.

**Comment by wangke19** [comment]
@jianlinliu, it got passed with 4.11, need your approval.

**Pull request merged**